File: 백준/Bronze/4597. 패리티/패리티.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function sol(){
  rl.question('', (input) => {
    let bit = input;
    if (input === '#'){
      rl.close();
      return;
    }
    let end = bit.slice(-1);
    let count = bit.split('1').length-1;
    let dap  = '';
    if (end === 'o') {
      if (count%2) {
        dap = bit.replace(/o/gi, '0');
      }
      else {
        dap = bit.replace(/o/gi, '1');
      }
    }
    else if (end === 'e') {
      if (count%2) {
        dap = bit.replace(/e/gi, '1');
      }
      else {
        dap = bit.replace(/e/gi, '0');
      }
    }
    console.log(dap);
    sol();
  });
}

sol();

File: 백준/Bronze/4597. 패리티/패리티.py
while 1:
    bit = input()
    if bit == '#':
        break
    if bit[-1] == 'o':
        # 홀수
        if bit.count('1')%2:
            bit = bit.replace('o', '0')
        else:
            bit = bit.replace('o', '1')
    elif bit[-1] == 'e':
        # 짝수
        if bit.count('1')%2:
            bit = bit.replace('e', '1')
        else:
            bit = bit.replace('e', '0')
    print(bit)